test/acceptance/features/bindAppToService.feature

```diff
@@ -217,3 +217,177 @@ Feature: Bind an application to a service
+                              - binding:env:object:secret:username
+                              - binding:env:object:secret:password
+                            - description: Name of the ConfigMap to hold the DB config
+                              displayName: DB Config Map
+                              path: configmap
+                              x-descriptors:
+                              - urn:alm:descriptor:io.kubernetes:ConfigMap
+                              - binding:env:object:configmap:db_host
+                              - binding:env:object:configmap:db_port
+            """
+        * The CRD "backservs.service.example.com" is present
+            """
+            apiVersion: apiextensions.k8s.io/v1beta1
+            kind: CustomResourceDefinition
+            metadata:
+                name: backservs.service.example.com
+            spec:
+                group: service.example.com
+                versions:
+                    - name: v1
+                      served: true
+                      storage: true
+                scope: Namespaced
+                names:
+                    plural: backservs
+                    singular: backserv
+                    kind: Backserv
+                    shortNames:
+                    - bs
+            """
+        * The CR "demo-backserv-cr-2" for kind "Backserv" is present
+            """
+            apiVersion: service.example.com/v1
+            kind: Backserv
+            metadata:
+                name: demo-backserv-cr-2
+            status:
+                secret: csv-demo-secret
+                configmap: csv-demo-cm
+            """
+        * The "configmap" "csv-demo-cm" is present
+            """
+            apiVersion: v1
+            kind: ConfigMap
+            metadata:
+                name: csv-demo-cm
+            data:
+                db_host: 172.72.2.0
+                db_port: "3306"
+            """
+        * The "secret" "csv-demo-secret" is present
+            """
+            apiVersion: v1
+            kind: Secret
+            metadata:
+                name: csv-demo-secret
+            type: Opaque
+            stringData:
+                username: admin
+                password: secret123
+            """
+        When Service Binding Request is applied to connect the database and the application
+            """
+            apiVersion: apps.openshift.io/v1alpha1
+            kind: ServiceBindingRequest
+            metadata:
+                name: sbr-csv-secret-cm-descriptors
+            spec:
+                backingServiceSelector:
+                    group: service.example.com
+                    version: v1
+                    kind: Backserv
+                    resourceRef: demo-backserv-cr-2
+            """
+        Then Secret "sbr-csv-secret-cm-descriptors" must contain data from resource type "secret" cr name "csv-demo-secret" and jsonpath "{.data.username}"
```

Comment:
We could make it simpler - we do not really care what is the source of  data in binding secret, we just care about its value. So how about

```
Secret X contains Y key with value Z 
```

We have this approach in https://github.com/redhat-developer/service-binding-operator/pull/597/files

Reply:
Yes, we can do it, but if you see it is tightly coupled with secrets data only which is base64 encoded whereas configmap and CR spec data will be in the plain string. So the specifying source will apprise about which data need to be encoded in base64 for the assertion. Which is what my step does. 
https://github.com/redhat-developer/service-binding-operator/pull/597/files#diff-ec891afbdb0cf5fd075cbf5ca535a45fR315

Reply:
I disagree - user do not care where the value originally comes - it is an implementation detail. From user perspective it is just important that given secret contains a key with some expected value. User do not need to know anything about encoding as well.

Reply:
Done.